flink-core/src/main/java/org/apache/flink/api/common/state/StateTtlConfig.java

```diff
@@ -433,6 +461,13 @@ public static class RocksdbCompactFilterCleanupStrategy
             implements CleanupStrategies.CleanupStrategy {
         private static final long serialVersionUID = 3109278796506988980L;
 
+        /**
+         * Default value lets RocksDB control this feature as needed. For now, RocksDB will change
+         * this value to 30 days (i.e 30 * 24 * 60 * 60) so that every file goes through the
+         * compaction process at least once every 30 days if not compacted sooner.
+         */
+        static final Time DEFAULT_PERIODIC_COMPACTION_TIME = Time.seconds(0xfffffffffffffffeL);
```

Comment:
I think we made a naive bug here. `0xfffffffffffffffeL` is actually `UINT64_MAX - 1`, which is is a special flag to allow RocksDB to pick default value.
This value does not equal to 2592000 seconds.

Reply:
I have used long type in the inner class to avoid the confusing semantics and also update the doc.
Please take a review agiain, Thanks!
I will squash some commits when you approve.

Reply:
As discussed offline, we will set the default value to 30 days in Flink to avoid the unpredictable change in rocksdb.